### PR TITLE
Utilities for handling Run-length-encoded Segmentation Maps

### PR DESCRIPTION
**Pull request by soumik12345**
# Utilities Added

- A function `rle_to_mask2d` is added that converts a Run-length encoded segmentation map to a 2-dimensional segmentation map. This function was adapted from [this StackOverflow answer](https://stackoverflow.com/a/58842473/8209968).
- A function `mask2d_to_rle` is added that converts a 2-dimensional segmentation map to a Run-length encoded segmentation map.

**Demo Notebook:** https://colab.research.google.com/drive/17ilyBukuAwwG1h7paB-JCGPmKP2zBqoV?usp=sharing

PR related to https://github.com/keras-team/keras-cv/issues/533 and https://github.com/keras-team/keras-cv/issues/327

CC: @LukeWood @tanzhenyu @sayakpaul @innat [@ayulockin](https://github.com/ayulockin)

### CODE REVIEW

**Comment by soumik12345** [comment]
CC: @sayakpaul

File: keras_cv/utils/rle_utils.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import numpy as np
```
Comment (sayakpaul) [inline keras_cv/utils/rle_utils.py:1]: Must be included with a license.

Example: https://github.com/keras-team/keras-cv/blob/master/keras_cv/layers/preprocessing/aug_mix.py#L1-#L13

File: keras_cv/utils/rle_utils.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import tensorflow as tf
+
+
+def rle_to_mask2d(mask_rle: str, shape) -> tf.Tensor:
```
Comment (sayakpaul) [inline keras_cv/utils/rle_utils.py:5]: We're not using type annotation at the moment.

File: keras_cv/utils/rle_utils.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    """Converts a Run-length encoded segmentation map to a 2-dimensional segmentation map.
+
+    Args:
+      mask_rle: a string that represents a run-length encoded segmentation map.
+      shape: shape of the resultant 2-dimensional segmentation map.
+
+    Returns:
+      mask: 2-dimensional segmentation map.
+    """
```
Comment (sayakpaul) [inline keras_cv/utils/rle_utils.py:14]: Include:

* A `Raises` flag (example) that discusses a corner case if that's known. 
* A sample example.

File: keras_cv/utils/rle_utils.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    shape = tf.convert_to_tensor(shape, tf.int64)
+    size = tf.math.reduce_prod(shape)
+    # Split string
+    s = tf.strings.split(mask_rle)
+    s = tf.strings.to_number(s, tf.int64)
```
Comment (sayakpaul) [inline keras_cv/utils/rle_utils.py:19]: When code-blocks are grouped please separate them with spaces.

File: keras_cv/utils/rle_utils.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+def mask2d_to_rle(mask: tf.Tensor) -> str:
+    """Converts a 2-dimensional segmentation map to a Run-length encoded segmentation map.
+
+    Args:
+      mask: a tensor that represents a 2-dimensional segmentation map.
+
+    Returns:
+      mask_rle: a string that represents the run-length encoded segmentation map.
+    """
```
Comment (sayakpaul) [inline keras_cv/utils/rle_utils.py:48]: Same. 

Include:

* `Raises` flag. 
* Sample usage.

**Review by sayakpaul — CHANGES_REQUESTED** [review]
@soumik12345 please include a test script as well. 

@LukeWood to do another pass.

File: keras_cv/utils/rle_utils.py [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import tensorflow as tf
+
+
+def rle_to_mask2d(mask_rle, shape):
```
Comment (tanzhenyu) [inline keras_cv/utils/rle_utils.py:19]: please add unit tests for these 2 functions

**Review by tanzhenyu — COMMENTED** [review]
Thanks for the PR!

**Comment by LukeWood** [comment]
Closing for now.  We may revisit this someday in the future, but for now we need to design an API for segmentation maps.

**Pull request closed without merging**